spec/takeaway_spec.rb

```diff
@@ -0,0 +1,65 @@
+require 'takeaway'
+
+describe Takeaway do
+
+  subject(:takeaway) { described_class.new(menu, order_klass, text) }
+  let(:menu) { double :menu, new: nil, dishes: dishes, on_menu?: nil }
+  let(:dishes) { {soup: 4.99, sandwich: 5.99, pie: 7.99} }
+  let(:order_klass) { double :order_klass, take_order: nil, new: order }
+  let(:order) { double :order, take_order: nil, current_order: current_order }
+  let(:current_order) { {pie: 3} }
+  let(:text) { double :text, send_text: nil }
+  let(:test_number) { double :test_nunber }
+
+  context 'when starting an order' do
+    describe '#check_menu' do
+      it 'should let them read the menu'do
+        expect(takeaway.check_menu).to include menu.dishes
+      end
+    end
+  end
+
+  context 'when placing an order' do
+    describe '#place_order' do
+
+    it 'should not let them order off-menu' do
+      expect{takeaway.place_order(:poison, 12)}.
+        to raise_error "Not on menu: poison"
+    end
+
+      it 'should update the customer on items in basket' do
+        line = "pie x 3: total £23.97\n"
+        expect{takeaway.place_order(:pie, 3)}.to output(line).to_stdout
+      end
+    end
+
+    describe '#check_order' do
+      it 'should let the customer view their current order' do
+        expect{takeaway.check_order}.to output("pie x 3: total £23.97\n").
+        to_stdout
```

Comment:
Style/MultilineOperationIndentation: Use 2 (not 0) spaces for indenting an expression spanning multiple lines.